File: src/com/magayaga/microscript/Scanner.java
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2025 Cyril John Magayaga
 * 
 * It was originally written in Java programming language.
 */
package com.magayaga.microscript;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

public class Scanner {
    private final String filePath;

    public Scanner(String filePath) {
        this.filePath = filePath;
    }

    public List<String> readLines() throws IOException {
        return Files.readAllLines(Paths.get(filePath));
    }
} 

File: src/microscript/Scanner.cs
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2026 Cyril John Magayaga
 * 
 * Converted from Java source.
 */
using System.Collections.Generic;
using System.IO;
namespace com.magayaga.microscript
{
    public class Scanner
    {
        private readonly string filePath;

        public Scanner(string filePath)
        {
            this.filePath = filePath;
        }

        public List<string> ReadLines()
        {
            return new List<string>(File.ReadAllLines(filePath));
        }
    }
}
